### DISCUSSION TITLE
Exclude file formats and paths

### DISCUSSION THREAD

**Discussion by Retengart** [mhx/dwarfs#227] [Q&A] [answered]
I read documentation for mkdwarfs, but a feeling tells me that I have not found a way to exclude certain types of files (or paths) from the source directory while creating dwarfs image. Any ideas?
For example, I want "to dwarfs" the root ( '/') and exclude /home directory

**Comment by mhx** [comment, marked as answer]
> I read documentation for mkdwarfs, but a feeling tells me that I have not found a way to exclude certain types of files (or paths) from the source directory while creating dwarfs image. Any ideas?

This should be doable using [filter rules](https://github.com/mhx/dwarfs/blob/main/doc/mkdwarfs.md#filter-rules). Did you see this when browsing the `mkdwarfs` docs? Did you see the `Filter options` section in the help output?

```
$ mkdwarfs -H
[...]
Filter options:
  -F [ --filter ] arg                   add filter rule
  --debug-filter [=arg(=all)]           show effect of filter rules without 
                                        producing an image (all, excluded, 
                                        excluded-files, files, included, 
                                        included-files)
  --remove-empty-dirs                   remove empty directories in file system
```

I'm genuinely curious why you failed to find that information. Which keywords would have helped you to find it? The documentation explicitly mentions both `include` and `exclude`, for example. Happy to improve the docs to make stuff like this easier to find!

> For example, I want "to dwarfs" the root ( '/') and exclude /home directory

That should be easy:

```
$ mkdwarfs -i / -o /home/root.dwarfs -F '- /home/'
```

Let me know if that helps!

**Reply by Retengart** [reply]
Thank you! I'm sorry for any inconvenience with this obviously trivial question. Surprisingly, when I searched the web page with "Ctrl + F" yesterday, the keywords "exclude" and "include" did not appear (maybe I got another accidental browser bug)
